packages/plugin-ext/src/plugin/type-converters.ts

```diff
@@ -705,6 +705,9 @@ export function fromTask(task: theia.Task): TaskDto | undefined {
     const taskDto = {} as TaskDto;
     taskDto.label = task.name;
     taskDto.source = task.source;
+    if (task.problemMatchers) {
+        taskDto.problemMatcher = task.problemMatchers;
+    }
```

Comment:
Hello!
I see you provided the changes for `fromTask()` function.
Should `toTask()` function have the corresponding changes?

Reply:
I mean `fromTask` and `toTask` functions are used for communication between `theia` and a plugin.
If `problemMatchers` were added to `fromTask` function - should we add some logic to `toTask ` function?
It makes sense to check that `problemMatchers` are not lost at converting from `TaskDto` to `theia.Task`.

`toTask ` function is used at resolving a task, for example.
I tried to check it using your extension, but looks like `resolveTask()` method is not executed for your custom resolver.

Reply:
I added `problemMatcher` to the destructured properties of TaskDto to avoid including it in the `taskDefinition`. But looking at what VS Code does, I think we shouldn't copy that to the Task at this point. I added the following code to my example project:

https://github.com/spoenemann/task-problem-matcher-example/blob/f48f5823afce15bc01dfc2c24b3d869d5dbe03a3/src/extension.ts#L25-L33

It prints the `problemMatchers` when `resolveTask` is called. I tested that with the following entry in `tasks.json`:

```json
{
  "version": "2.0.0",
  "tasks": [
    {
      "type": "mytask",
      "label": "Foo",
      "problemMatcher": "$mymatcher"
    }
  ]
}
```

When running that task, VS Code shows a message "problemMatchers: []", so the value of the `problemMatcher` property is _not_ passed to the task resolver. One reason for this might be that the properties are incompatible in this direction. Here's the description of `problemMatcher` in tasks.json:

> The problem matcher(s) to use. Can either be a string or a problem matcher definition or an array of strings and problem matchers.

This matches what we have in our `TaskCustomization` interface.

When I tried to run that task in Theia, I found another problem (unrelated to this change): the task resolver is not called, or at least I don't see any info message popping up. Is this a known problem?

Reply:
I'll test it today and write the results here

Reply:
I investigated the problem:
- task system resolves a task using registered resolvers
- task service takes the corresponding resolver by task type
- the problem is: the custom task from the extension comes with type `shell`, `mytask` type is present as `taskType` field in the configuration
- as result - `ProcessTaskResolver` is used for resolving the task, not custom resolver from the extension

![resolver](https://user-images.githubusercontent.com/5676062/99879229-19474000-2c14-11eb-8fbe-e77796365ff7.png)

Reply:
About VS Code behavior - it's really strange for me, because it looks like resolving a task works for me if I run a task from `recently used` section only - resolving a task doesn't work if I do it from `contributed` section

![VSCode_resolve](https://user-images.githubusercontent.com/5676062/99879351-2fa1cb80-2c15-11eb-8042-c3f131b79735.gif)